### PR TITLE
add value protection

### PR DESCRIPTION
**Pull request by ChristophTobler**
For time to time the gyro and dt values are way to big. I added a protection against that. You can either merge this PR or https://github.com/intel-aero/aero-optical-flow/pull/16 and then I'll rebase.

### CODE REVIEW

File: src/mainloop.cpp [hunk 1]
```diff
@@ -412,13 +412,22 @@ static void _highres_imu_msg_callback(const mavlink_highres_imu_t *msg, void *da
...
-		_gyro_integrated.x += (msg->xgyro * t);
-		_gyro_integrated.y += (msg->ygyro * t);
-		_gyro_integrated.z += (msg->zgyro * t);
+	const double dt = (double)(msg->time_usec - _gyro_prev_timestamp) / (double)USEC_PER_SEC;
+
+	// protect against invalid gyro and timestamp data -> huge random values
+  const double max_dt = 0.05; // at least 20Hz
+  const double max_gyro = 20.0; // big enough value
+	if (_gyro_prev_timestamp &&
+	    dt < max_dt && abs(msg->xgyro) < max_gyro && abs(msg->ygyro) < max_gyro && abs(msg->zgyro) < max_gyro) {
```
Comment (zehortigoza) [inline src/mainloop.cpp]: Looks good but the code style is complete messed.

File: src/mainloop.cpp [hunk 2]
```diff
@@ -506,7 +515,7 @@ void Mainloop::shutdown()
 #else
 	delete _mavlink;
 #endif
-
```
Comment (zehortigoza) [inline src/mainloop.cpp]: remove it

**Review by zehortigoza — CHANGES_REQUESTED** [review]
Please fix the code style and just send the 'add protection against huge gyro/dt values' patch.
@avinash-palleti can pickup your 'CMakeLists: print msg when using FastRTPS' and add to his PR.

**Comment by ChristophTobler** [comment]
@zehortigoza done

**Comment by zehortigoza** [comment]
merged thanks

**Pull request closed without merging**